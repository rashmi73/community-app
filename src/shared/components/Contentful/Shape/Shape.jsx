/**
 * The core shape rendering.
 */

import PT from 'prop-types';
import React from 'react';
import _ from 'lodash';
import { themr } from 'react-css-super-themr';
import { fixStyle } from 'utils/contentful';
import SVG from 'react-inlinesvg';

import LoadingIndicator from 'components/LoadingIndicator';
import defaultTheme from './themes/default.scss';

export function ShapeInner({
  theme,
  id,
  shape,
  shapeSvg,
}) {
  const {
    height,
    backgroundColor,
    inline,
  } = shape;

  const shapeSvgUrl = _.get(shapeSvg, 'file.url');
  const style = fixStyle(shape.extraStylesForContainer) || {};
  if (!shapeSvgUrl) {
    return (
      <div
        id={id}
        className={theme['shape-wrap']}
        style={style}
      />
    );
  }
  if (!inline) {
    style.backgroundImage = `url(${shapeSvgUrl})`;
  }
  if (height) {
    style.height = height;
  }
  if (backgroundColor) {
    style.backgroundColor = backgroundColor;
  }
  return (
    <div
      id={id}
      className={theme['shape-wrap']}
      style={style}
    >
      {inline && (
        <SVG
          preloader={<LoadingIndicator />}
          src={shapeSvgUrl}
        />
      )}
    </div>
  );
}

ShapeInner.defaultProps = {
  shapeSvg: null,
};

ShapeInner.propTypes = {
  id: PT.string.isRequired,
  shape: PT.shape().isRequired,
  shapeSvg: PT.shape(),
  theme: PT.shape({
    'shape-wrap': PT.string,
  }).isRequired,
};

export default themr('Shape', defaultTheme)(ShapeInner);
