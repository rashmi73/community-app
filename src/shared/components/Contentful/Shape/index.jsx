/**
 * Shape component.
 */

import ContentfulLoader from 'containers/ContentfulLoader';
import LoadingIndicator from 'components/LoadingIndicator';
import PT from 'prop-types';
import React from 'react';
import _ from 'lodash';

import Shape from './Shape';
import defaultTheme from './themes/default.scss';


/* Loads the main shape entry. */
export default function ShapeLoader(props) {
  const {
    id, preview, spaceName, environment,
  } = props;
  return (
    <ContentfulLoader
      entryIds={id}
      preview={preview}
      spaceName={spaceName}
      environment={environment}
      render={(data) => {
        const { fields } = data.entries.items[id];
        const shapeSvgId = _.get(fields, 'shapeSvg.sys.id');
        const assetIds = _.compact([shapeSvgId]);
        if (assetIds.length !== 0) {
          return (
            <ContentfulLoader
              assetIds={assetIds}
              preview={preview}
              spaceName={spaceName}
              environment={environment}
              render={(assetData) => {
                const { items } = assetData.assets;
                const shapeSvgFields = _.get(items, [shapeSvgId, 'fields']);
                return (
                  <Shape
                    {...props}
                    shapeSvg={shapeSvgFields}
                    shape={fields}
                    theme={defaultTheme}
                  />
                );
              }}
              renderPlaceholder={LoadingIndicator}
            />
          );
        }
        return <Shape {...props} shape={fields} theme={defaultTheme} />;
      }}
      renderPlaceholder={LoadingIndicator}
    />
  );
}

ShapeLoader.defaultProps = {
  preview: false,
  spaceName: null,
  environment: null,
};

ShapeLoader.propTypes = {
  id: PT.string.isRequired,
  preview: PT.bool,
  spaceName: PT.string,
  environment: PT.string,
};
