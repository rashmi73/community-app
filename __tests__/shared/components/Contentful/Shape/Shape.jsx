import React from 'react';
import Renderer from 'react-test-renderer/shallow';
import { ShapeInner } from 'components/Contentful/Shape/Shape';

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();
  const MOCK_PROPS = {
    id: '1',
    shape: {},
    shapeSvg: {
      file: {
        url: 'https://www.topcoder.com',
      },
    },
    theme: {},
  };
  renderer.render(<ShapeInner {...MOCK_PROPS} />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
