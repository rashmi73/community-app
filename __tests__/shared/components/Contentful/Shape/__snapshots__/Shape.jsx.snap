// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<div
  id="1"
  style={
    Object {
      "backgroundImage": "url(https://www.topcoder.com)",
    }
  }
/>
`;
